components/applications-service/integration_test/services_test.go

```diff
@@ -162,6 +173,49 @@ func TestGetServicesMultiService(t *testing.T) {
 	assertServicesEqual(t, expected.GetServices(), response.GetServices())
 }
 
+func TestGetServicesWithDisconnectedServices(t *testing.T) {
+	inputSvcs := habServicesMatrixAllHealthStatusDifferent()
+
+	// Sending the timestamp roundtrip through the app to postgres and back
+	// removes some resolution which makes the timestamps not compare exactly.
+	// Rounding the time we send to an even number of seconds fixes the issue.
+	eventTime := time.Now().Add(-time.Minute * 30).Round(time.Second)
+
+	for _, s := range inputSvcs {
+		var err error
+		s.EventMetadata.OccurredAt, err = ptypes.TimestampProto(eventTime)
+		require.NoError(t, err)
+	}
+
+	suite.IngestServices(inputSvcs)
+	defer suite.DeleteDataFromStorage()
+
+	_, err := suite.ApplicationsServer.MarkDisconnectedServices(300)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	req := &applications.ServicesReq{}
+	res, err := suite.ApplicationsServer.GetServices(ctx, req)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 6, len(res.Services))
+	for _, s := range res.Services {
+		assert.True(t, s.Disconnected, "expected the service to be marked as disconnected")
+		expectedTime, err := ptypes.TimestampProto(eventTime)
+		require.NoError(t, err)
+		assert.Equal(t, expectedTime, s.LastEventOccurredAt)
+		// Ideally we could do:
+		//   assert.Equal(t, "30 minutes", s.LastEventSince)
+		// but we _usually_ get something like "29 minutes, 59 seconds" because we
```

Comment:
Would [`WithinDuration`](https://godoc.org/github.com/stretchr/testify/assert#WithinDuration) help?

Reply:
Not quite, we have code within the API that makes a string (kinda like a similar helper function in ruby on rails) with the relative time in words, I'm checking the content of that string.

Reply:
Ah now I get it. I guess you could parse the duration string in the test... But probably not worth it. ;)